packages/sdk/client/src/client/client.ts

```diff
@@ -239,26 +239,63 @@ export class Client {
    */
   async repair(): Promise<any> {
     // TODO(burdon): Factor out.
-    const spaces = this.spaces.get();
-    const docs = spaces
-      .map((space) =>
-        (space as any)._data.pipeline.currentEpoch?.subject.assertion.automergeRoot.slice('automerge:'.length),
-      )
-      .filter(Boolean);
-
-    let removed = 0;
-    if (typeof navigator !== 'undefined' && navigator.storage) {
-      const dir = await navigator.storage.getDirectory();
-      for await (const filename of (dir as any)?.keys()) {
-        if (filename.includes('automerge_') && !docs.some((doc) => filename.includes(doc))) {
-          await dir.removeEntry(filename);
-          removed++;
+    const repairSummary: any = {};
+
+    {
+      // Cleanup OPFS.
+      const spaces = this.spaces.get();
+      const docs = spaces
+        .map((space) =>
+          (space as any)._data.pipeline.currentEpoch?.subject.assertion.automergeRoot.slice('automerge:'.length),
+        )
+        .filter(Boolean);
+
+      repairSummary.OPFSRemovedFiles = 0;
+      if (typeof navigator !== 'undefined' && navigator.storage) {
+        const dir = await navigator.storage.getDirectory();
+        for await (const filename of (dir as any)?.keys()) {
+          if (filename.includes('automerge_') && !docs.some((doc) => filename.includes(doc))) {
+            await dir.removeEntry(filename);
+            repairSummary.OPFSRemovedFiles++;
+          }
         }
       }
     }
 
-    log.info('Repair succeeded', { removed });
-    return { removed };
+    {
+      // Fix storage config.
+      const config = {
+        runtime: {
+          client: {
+            storage: {
+              dataStore: this.config.values.runtime?.client?.storage?.dataStore,
+            },
+          },
+        },
+      };
+      await SaveConfig(config);
+
+      repairSummary.storageConfig = config;
+    }
+
+    {
+      repairSummary.levelDBRemovedEntries = 0;
+      // Cleanup old index-data from level db.
+      const level = await createLevel(this._config?.values.runtime?.client?.storage ?? {});
```

Comment:
Separatelly: level has a pre-write hook. Can you add validation code that will make sure we're only writing to correct sublevels